StackableGems/mod.js

```diff
@@ -151,26 +151,59 @@ for (let i = 0; i < ITEM_TYPES.length; i = i + 1) {
 
 // this is behind a config option because it's *a lot* of recipes...
 if (config.convertWhenDestacking) {
-  for (let i = 0; i < ITEM_TYPES.length; i = i + 1) {
-    const itemtype = ITEM_TYPES[i];
-    const stacktype = converItemTypeToStackItemType(itemtype);
-    for (let j = 2; j <= 500; j = j + 1) {
+  for (let i = 0; i < ITEM_TYPES.length - 1; i = i + 1) {
+    const itemtype_cur = ITEM_TYPES[i];
+    const stacktype_cur = converItemTypeToStackItemType(itemtype_cur);
+    const itemtype_up = ITEM_TYPES[i + 1];
+    const stacktype_up = converItemTypeToStackItemType(itemtype_up);
+    for (let j = 2; j <= config.maxInStack; j = j + 1) {
       cubemain.rows.push({
-        description: `Stack of ${j} ${itemtype} -> Stack of ${
-          j - 1
-        } and Stack of 1`,
+        description: `Stack of ${j} ${stacktype_cur} -> Stack of ${j - 1} ${stacktype_cur} and ${itemtype_cur}`,
         enabled: 1,
         version: 0,
         op: 18, // skip recipe if item's Stat.Accr(param) != value
         param: 70, // quantity (itemstatcost.txt)
         value: j, // only execute rule if quantity == j
         numinputs: 1,
-        'input 1': stacktype,
-        output: `"${stacktype},qty=${j - 1}"`,
-        'output b': `"${itemtype},qty=1"`,
+        'input 1': stacktype_cur,
+        output: `"${stacktype_cur},qty=${j - 1}"`,
+        'output b': `"${itemtype_cur},qty=1"`,
         '*eol': 0,
       });
     }
+    if (config.upgradeStack) {      
+      cubemain.rows.push({
+        description: `Stack of 2 ${stacktype_cur} and ${itemtype_cur} -> Stack of 1 ${stacktype_up}`,
+        enabled: 1,
+        version: 0,
+        op: 18, // skip recipe if item's Stat.Accr(param) != value
+        param: 70, // quantity (itemstatcost.txt)
+        value: 2, // only execute rule if quantity == j
+        numinputs: 2,
+        'input 1': stacktype_cur,
+        'input 2': itemtype_cur,
+        'output': `"${stacktype_up},qty=1"`,
+        '*eol': 0,
+      });
+      for (let j = 3; j <= config.maxInStack; j = j + 1) {
+        cubemain.rows.push({
+          description: `Stack of ${j} ${stacktype_cur} and ${itemtype_cur} -> Stack of ${
+            j - 2
+          } ${stacktype_cur} and Stack of 1 ${stacktype_up}`,
+          enabled: 1,
+          version: 0,
+          op: 18, // skip recipe if item's Stat.Accr(param) != value
+          param: 70, // quantity (itemstatcost.txt)
+          value: j, // only execute rule if quantity == j
+          numinputs: 2,
+          'input 1': stacktype_cur,
+          'input 2': itemtype_cur,
+          output: `"${stacktype_cur},qty=${j - 2}"`,
+          'output b': `"${stacktype_up},qty=1"`,
+          '*eol': 0,
+        });
```

Comment:
The base case does not need to be duplicate code. e.g.:

```suggestion
      for (let j = 2; j <= config.maxInStack; j = j + 1) {
        cubemain.rows.push({
          description: `Stack of ${j} ${stacktype_cur} and ${itemtype_cur} ->`
            + (j >= 3 ? ` Stack of ${j - 2} ${stacktype_cur}` : '')
            + ` and Stack of 1 ${stacktype_up}`,
          enabled: 1,
          version: 0,
          op: 18, // skip recipe if item's Stat.Accr(param) != value
          param: 70, // quantity (itemstatcost.txt)
          value: j, // only execute rule if quantity == j
          numinputs: 2,
          'input 1': stacktype_cur,
          'input 2': itemtype_cur,
          output: `"${stacktype_up},qty=1"`,
          'output b': j >= 3 ? `"${stacktype_cur},qty=${j - 2}"` : null,
          '*eol': 0,
        });
```